test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -319,3 +319,206 @@
+                         (diffs "2022-10-02T01:00:00+01:00"      ; 2022-10-02T00:00:00+00:00
+                                "2023-10-02T00:00:00+00:00"))))) ; 2023-10-02T00:00:00+00:00
+        (testing "hour under a year"
+          (mt/with-temporary-setting-values [driver/report-timezone "Atlantic/Cape_Verde"]
+            (is (partial= {:day 365 :month 12 :year 1}
+                          (diffs "2022-10-02T00:00:00+00:00"     ; 2022-10-01T23:00:00-01:00
+                                 "2023-10-02T00:00:00+01:00")))) ; 2023-10-01T22:00:00-01:00
+          (mt/with-temporary-setting-values [driver/report-timezone "UTC"]
+            (is (partial= {:day 364 :month 11 :year 0}
+                          (diffs "2022-10-02T00:00:00+00:00"          ; 2022-10-02T00:00:00+00:00
+                                 "2023-10-02T00:00:00+01:00"))))))))) ; 2023-10-01T23:00:00+00:00
+
+(deftest datetime-diff-expressions-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :datetime-diff)
+    (mt/dataset sample-dataset
+      (testing "Args can be expressions that return datetime values"
+        (let [diffs (fn [x y]
+                      (let [units [:second :minute :hour :day :week :month :year]]
+                        (->> (mt/run-mbql-query orders
+                               {:limit 1
+                                :expressions (into {} (for [unit units]
+                                                        [(name unit) [:datetime-diff x y unit]]))
+                                :fields (into [] (for [unit units]
+                                                   [:expression (name unit)]))})
+                             mt/rows first
+                             (zipmap units))))]
+          (is (= {:second 31795200, :minute 529920, :hour 8832, :day 368, :week 52, :month 12, :year 1}
+                 (diffs [:datetime-add #t "2022-10-03T00:00:00" 1 "day"] [:datetime-add #t "2023-10-03T00:00:00" 4 "day"])))))
+      (testing "Result works in arithmetic expressions"
+        (let [start "2021-10-03T09:19:09"
+              end   "2022-10-03T09:18:09"]
+          (is (= [[1 6 365 370]]
+                 (mt/rows
+                  (mt/run-mbql-query orders
+                    {:limit       1
+                     :expressions {"datediff1"     [:datetime-diff start end :year]
+                                   "datediff1-add" [:+ [:datetime-diff start end :year] 5]
+                                   "datediff2"     [:datetime-diff start end :day]
+                                   "datediff2-add" [:+ 5 [:datetime-diff start end :day]]}
+                     :fields      [[:expression "datediff1"]
+                                   [:expression "datediff1-add"]
+                                   [:expression "datediff2"]
+                                   [:expression "datediff2-add"]]})))))))))
+
+(deftest datetime-diff-type-test
+  (mt/test-drivers #{:bigquery-cloud-sdk}
```

Comment:
```suggestion
  (mt/test-drivers (mt/normal-drivers-with-feature :datetime-diff)
```

Reply:
I plan to implement this in a subsequent PR. Is that OK?